### PR TITLE
Add CIFAR10 to CI

### PR DESCRIPTION
**Pull request by TolyaTalamanov**

### CODE REVIEW

**Comment by TolyaTalamanov** [comment]
@NeuralDeepWorks/maintainers Have a look, please

File: samples/cifar10/README.md [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+* Follow instruction to setup enviroment: [DeepWorks samples](../README.md)
+
+* Run train:
+```bash
+./samples/mnist/run_cifar10_train.sh
+```
+
+Possible output:
+```bash
+// TODO
```
Comment (TolyaTalamanov) [inline samples/cifar10/README.md:12]: I'll put after CI finishes

File: tests/unit/test_utils.hpp [hunk 2]
```diff
@@ -40,11 +40,12 @@ inline void AssertEqual(const deepworks::Tensor& actual,
 inline void AssertTensorEqual(const deepworks::Tensor& actual,
                               const deepworks::Tensor& expected,
                               float threshold = 1e-5) {
+    // FIXME: Deprecated approach will be removed in future.
     ASSERT_EQ(actual.shape()  , expected.shape());
     ASSERT_EQ(actual.strides(), expected.strides());
 
     float norm = normL1(actual, expected);
-    EXPECT_LE(norm, 1e-5);
+    EXPECT_LE(norm, 1e-3);
```
Comment (TolyaTalamanov) [inline tests/unit/test_utils.hpp:48]: @l-bat We've discussed that it can be `1e-2`, but I believe stronger threshold should work as well

Comment (l-bat) [reply tests/unit/test_utils.hpp:47]: Could you try to change it to 1e-4?

**Review by l-bat — APPROVED** [review]

File: samples/cifar10/README.md [hunk 3]
```diff
@@ -1,16 +1,40 @@
...
+
+Possible output:
+```bash
+Epoch: 0
+Loss: 2.21671
+Loss: 2.04742
+Loss: 1.92269
+Loss: 1.91002
+Accuracy: 0.343249
+Model saved: build/cifar10_model.bin
```
Comment (RodionovF) [inline samples/cifar10/README.md:18]: if you need it, i can give you more more extensive output with Accuracy after ~40 epochs, but a few hours later

like

```
Epoch: 0
Loss: 2.21671
Loss: 2.04742
Loss: 1.92269
Loss: 1.91002
Accuracy: 0.343249

...

Epoch: 26
Loss: 1.04711
Loss: 1.05685
Loss: 1.05886
Loss: 1.11385
Loss: 1.14189
Loss: 1.14818
Loss: 1.14877
Accuracy: +100500
```

**Comment by RodionovF** [comment]
can you change optimizer in cifar10/main.cpp to

```dw::optimizer::Adam opt(model.params(), 1e-3); ```?

I did a rebase yesterday and forgot it

**Comment by RodionovF** [comment]
good job, lgtm :+1:

File: samples/cifar10/README.md (hunk 3 above)
Comment (TolyaTalamanov) [reply samples/cifar10/README.md:18]: I believe it's OK for README.md but would be really interesting to know actual results

File: tests/unit/test_utils.hpp (hunk 2 above)
Comment (TolyaTalamanov) [reply tests/unit/test_utils.hpp]: Done

**Pull request merged**